lib/child_process.js

```diff
@@ -276,31 +273,30 @@ ObjectDefineProperty(exec, promisify.custom, {
  *   ) => any} [callback]
  * @returns {ChildProcess}
  */
-function execFile(file /* , args, options, callback */) {
-  let args = [];
-  let callback;
-  let options;
-
-  // Parse the optional positional parameters.
-  let pos = 1;
-  if (pos < arguments.length && ArrayIsArray(arguments[pos])) {
-    args = arguments[pos++];
-  } else if (pos < arguments.length && arguments[pos] == null) {
-    pos++;
-  }
-
-  if (pos < arguments.length && typeof arguments[pos] === 'object') {
-    options = arguments[pos++];
-  } else if (pos < arguments.length && arguments[pos] == null) {
-    pos++;
+function execFile(file, args = [], options, callback) {
+  if (args == null) {
+    args = [];
+  } else if (typeof args === 'object') {
+    if (!ArrayIsArray(args)) {
+      callback = options;
+      options = args;
+      args = [];
+    }
+  } else if (typeof args === 'function') {
+    callback = args;
+    options = null;
+    args = [];
   }
 
-  if (pos < arguments.length && typeof arguments[pos] === 'function') {
-    callback = arguments[pos++];
+  if (typeof options === 'function') {
```

Comment:
```suggestion
  else if (typeof options === 'function') {
```

Reply:
The proposed change here will work, but I wonder about maintainability. The way it's written now, there's an if/else to deal with `args`, then another if/else to deal with `options`, and finally an if/else to deal with `callback`.  This is both easy to follow cognitively and makes sense because while processing `args`, the code may reassign `options` and `callback` to different values. Right now, there's no way for `options` to be assigned to a function, but if something changed and that became a possibility during `args` processing, this code will still work. If we add the `else` here, we get rid of that separation and increase the possibility of an error being introduced by a change in seemingly unrelated code.

Reply:
> Using `else` avoids rechecking `options` value that was changed in the previous step, without it the engine is forced to recheck it several times. Not sure it actually matters though.

Yeah, the trade off is maintainability/readability vs. efficiency/performance. In both cases, the advantages are pretty small, I think. I have my opinion, but it's not a strong one. Based on your "Not sure it actually matters though", it sounds like you similarly have an opinion, but not a strong one.